### ISSUE TITLE
[Bug] Broken replication

### ISSUE THREAD

**Issue by sle78** [mariadb-operator/mariadb-operator#1015]
**Documentation**
- [x] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**

`kubectl describe mariadbs testrail`

```
Status:
  Conditions:
    Last Transition Time:     2024-10-31T18:25:13Z
    Message:                  Running
    Reason:                   StatefulSetReady
    Status:                   True
    Type:                     Ready
    Last Transition Time:     2024-10-31T18:24:33Z
    Message:                  Updated
    Reason:                   Updated
    Status:                   True
    Type:                     Updated
  Current Primary:            testrail-db-0
  Current Primary Pod Index:  0
  Replicas:                   2
  Replication Status:
    testrail-db-0:  Master
    testrail-db-1:  Slave
Events:             <none>
```

```sql
MariaDB [testrail]> SHOW ALL REPLICAS STATUS\G
*************************** 1. row ***************************
               Connection_name: mariadb-operator
               Slave_SQL_State: Write_rows_log_event::write_row(-1) on table `cases`
                Slave_IO_State: Waiting for master to send event
                   Master_Host: testrail-db-0.testrail-db-internal
                   Master_User: repl
                   Master_Port: 3306
                 Connect_Retry: 10
               Master_Log_File: testrail-db-bin.000017
           Read_Master_Log_Pos: 805821
                Relay_Log_File: testrail-db-relay-bin-mariadb@002doperator.000008
                 Relay_Log_Pos: 474333823
         Relay_Master_Log_File: testrail-db-bin.000005
              Slave_IO_Running: Yes
             Slave_SQL_Running: Yes
               Replicate_Do_DB: 
           Replicate_Ignore_DB: 
            Replicate_Do_Table: 
        Replicate_Ignore_Table: 
       Replicate_Wild_Do_Table: 
   Replicate_Wild_Ignore_Table: 
                    Last_Errno: 0
                    Last_Error: 
                  Skip_Counter: 0
           Exec_Master_Log_Pos: 993450280
               Relay_Log_Space: 6397494372
               Until_Condition: None
                Until_Log_File: 
                 Until_Log_Pos: 0
            Master_SSL_Allowed: Yes
            Master_SSL_CA_File: 
            Master_SSL_CA_Path: 
               Master_SSL_Cert: 
             Master_SSL_Cipher: 
                Master_SSL_Key: 
         Seconds_Behind_Master: 724090
 Master_SSL_Verify_Server_Cert: Yes
                 Last_IO_Errno: 0
                 Last_IO_Error: 
                Last_SQL_Errno: 0
                Last_SQL_Error: 
   Replicate_Ignore_Server_Ids: 
              Master_Server_Id: 10
                Master_SSL_Crl: 
            Master_SSL_Crlpath: 
                    Using_Gtid: Current_Pos
                   Gtid_IO_Pos: 0-10-519547
       Replicate_Do_Domain_Ids: 
   Replicate_Ignore_Domain_Ids: 
                 Parallel_Mode: optimistic
                     SQL_Delay: 0
           SQL_Remaining_Delay: NULL
       Slave_SQL_Running_State: Write_rows_log_event::write_row(-1) on table `cases`
              Slave_DDL_Groups: 231
Slave_Non_Transactional_Groups: 184
    Slave_Transactional_Groups: 2137
          Replicate_Rewrite_DB: 
          Retried_transactions: 0
            Max_relay_log_size: 1073741824
          Executed_log_entries: 274798
     Slave_received_heartbeats: 0
        Slave_heartbeat_period: 30.000
                Gtid_Slave_Pos: 0-10-2551
```

**Expected behaviour**

**Steps to reproduce the bug**

1. ...
2. ...
3. ...

**Debug information**
master logs:
```
2024-10-31 18:25:13 17 [Note] Semi-sync replication enabled on the master.
2024-10-31 18:25:13 0 [Note] Starting ack receiver thread
2024-10-31 18:25:13 19 [Note] Start binlog_dump to slave_server(11), pos(, 4), using_gtid(1), gtid('')
2024-10-31 18:25:13 19 [Note] Start semi-sync binlog_dump to slave (server_id: 11), pos(./testrail-db-bin.000002, 285)
2024-10-31 18:25:23 20 [Warning] Timeout waiting for reply of binlog (file: testrail-db-bin.000017, pos: 1152), semi-sync up to file testrail-db-bin.000002, position 285.
2024-10-31 18:25:23 20 [Note] Semi-sync replication switched OFF.
2024-10-31 18:26:00 0 [Note] InnoDB: Buffer pool(s) load completed at 241031 18:26:00
2024-10-31 18:26:43 78 [ERROR] Invalid (old?) table or database name 'lost+found'
2024-10-31 18:26:51 78 [ERROR] Invalid (old?) table or database name 'lost+found'
2024-10-31 18:40:19 396 [Warning] Aborted connection 396 to db: 'unconnected' user: 'unauthenticated' host: '192.168.3.82' (This connection closed normally without authentication)
2024-10-31 18:49:12 780 [ERROR] Invalid (old?) table or database name 'lost+found'
```

replica logs:
```
2024-10-31 18:24:35 0 [Note] Starting MariaDB 11.4.3-MariaDB-ubu2404-log source revision 5ab81ffe0097a22a774957df28c5223cf0201de3 server_uid pSx1moX51HqKDIx5yiptMlpkIWY= as process 1
2024-10-31 18:24:36 0 [Note] InnoDB: Compressed tables use zlib 1.3
2024-10-31 18:24:36 0 [Note] InnoDB: Number of transaction pools: 1
2024-10-31 18:24:36 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2024-10-31 18:24:36 0 [Note] mariadbd: O_TMPFILE is not supported on /tmp (disabling future attempts)
2024-10-31 18:24:36 0 [Note] InnoDB: Initializing buffer pool, total size = 1.000GiB, chunk size = 16.000MiB
2024-10-31 18:24:36 0 [Note] InnoDB: Completed initialization of buffer pool
2024-10-31 18:24:36 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
2024-10-31 18:24:36 0 [Note] InnoDB: End of log at LSN=74192
2024-10-31 18:24:36 0 [Note] InnoDB: Opened 3 undo tablespaces
2024-10-31 18:24:36 0 [Note] InnoDB: 128 rollback segments in 3 undo tablespaces are active.
2024-10-31 18:24:36 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2024-10-31 18:24:36 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2024-10-31 18:24:36 0 [Note] InnoDB: log sequence number 74192; transaction id 45
2024-10-31 18:24:36 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-10-31 18:24:36 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2024-10-31 18:24:36 0 [Note] Plugin 'wsrep-provider' is disabled.
2024-10-31 18:24:36 0 [Note] InnoDB: Buffer pool(s) load completed at 241031 18:24:36
2024-10-31 18:24:41 0 [Note] Server socket created on IP: '0.0.0.0'.
2024-10-31 18:24:41 0 [Note] Server socket created on IP: '::'.
2024-10-31 18:24:41 0 [Note] mariadbd: Event Scheduler: Loaded 0 events
2024-10-31 18:24:41 0 [Note] mariadbd: ready for connections.
Version: '11.4.3-MariaDB-ubu2404-log'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
2024-10-31 18:25:13 17 [Note] Master connection name: 'mariadb-operator'  Master_info_file: 'master-mariadb@002doperator.info'  Relay_info_file: 'relay-log-mariadb@002doperator.info'
2024-10-31 18:25:13 17 [Note] 'CHANGE MASTER TO executed'. Previous state master_host='', master_port='3306', master_log_file='', master_log_pos='4'. New state master_host='testrail-db-0.testrail-db-internal', master_port='3306', master_log_file='', master_log_pos='4'.
2024-10-31 18:25:13 17 [Note] Previous Using_Gtid=Slave_Pos. New Using_Gtid=Current_Pos
2024-10-31 18:25:13 18 [Note] Master 'mariadb-operator': Slave I/O thread: Start semi-sync replication to master 'repl@testrail-db-0.testrail-db-internal:3306' in log '' at position 4
2024-10-31 18:25:13 19 [Note] Master 'mariadb-operator': Slave SQL thread initialized, starting replication in log 'FIRST' at position 4, relay log './testrail-db-relay-bin-mariadb@002doperator.000001' position: 4; GTID position ''
2024-10-31 18:25:13 18 [Note] Master 'mariadb-operator': Slave I/O thread: connected to master 'repl@testrail-db-0.testrail-db-internal:3306',replication starts at GTID position ''

```
- Related object events:
```bash
kubectl get events --field-selector involvedObject.name=<mariadb-resource-name>
kubectl get events --field-selector involvedObject.name=<backup-resource-name>
kubectl get events --field-selector involvedObject.name=<restore-resource-name>
```
- `mariadb-operator` logs. Set the `--log-level` to `debug` if needed.

**Environment details**:
- Kubernetes version: 1.29.10
- Kubernetes distribution: Vanilla
- mariadb-operator version: 0.35.1
- Galera data-plane version (init and agent): [Version number]
- Install method: helm

**Additional context**
```
MariaDB [testrail]> show status like "%slave%";
+------------------------------+--------+
| Variable_name                | Value  |
+------------------------------+--------+
| Com_show_slave_hosts         | 0      |
| Com_show_slave_status        | 4      |
| Com_start_all_slaves         | 0      |
| Com_start_slave              | 1      |
| Com_stop_all_slaves          | 0      |
| Com_stop_slave               | 0      |
| Rpl_semi_sync_slave_send_ack | 1851   |
| Rpl_semi_sync_slave_status   | ON     |
| Slave_connections            | 0      |
| Slave_heartbeat_period       | 30.000 |
| Slave_open_temp_tables       | 0      |
| Slave_received_heartbeats    | 0      |
| Slave_retried_transactions   | 0      |
| Slave_running                | OFF    |
| Slave_skipped_errors         | 0      |
| Slaves_connected             | 0      |
| Slaves_running               | 1      |
+------------------------------+--------+

```
```
MariaDB [testrail]> show status like "%master%";
+--------------------------------------------+---------+
| Variable_name                              | Value   |
+--------------------------------------------+---------+
| Com_change_master                          | 0       |
| Innodb_master_thread_active_loops          | 0       |
| Innodb_master_thread_idle_loops            | 4842    |
| Master_gtid_wait_count                     | 0       |
| Master_gtid_wait_time                      | 0       |
| Master_gtid_wait_timeouts                  | 0       |
| Rpl_semi_sync_master_clients               | 1       |
| Rpl_semi_sync_master_get_ack               | 1887    |
| Rpl_semi_sync_master_net_avg_wait_time     | 0       |
| Rpl_semi_sync_master_net_wait_time         | 0       |
| Rpl_semi_sync_master_net_waits             | 1887    |
| Rpl_semi_sync_master_no_times              | 1       |
| Rpl_semi_sync_master_no_tx                 | 928     |
| Rpl_semi_sync_master_request_ack           | 1888    |
| Rpl_semi_sync_master_status                | ON      |
| Rpl_semi_sync_master_timefunc_failures     | 0       |
| Rpl_semi_sync_master_tx_avg_wait_time      | 1885    |
| Rpl_semi_sync_master_tx_wait_time          | 3554486 |
| Rpl_semi_sync_master_tx_waits              | 1885    |
| Rpl_semi_sync_master_wait_pos_backtraverse | 0       |
| Rpl_semi_sync_master_wait_sessions         | 0       |
| Rpl_semi_sync_master_yes_tx                | 1886    |
+--------------------------------------------+---------+
```

**Comment by sle78** [comment]
Tried to delete the replica and its PVC:

```
2024-10-31 19:54:13 0 [Note] Recovering after a crash using testrail-db-bin
2024-10-31 19:54:13 0 [Note] Starting table crash recovery...
2024-10-31 19:54:13 0 [Note] Crash table recovery finished.
2024-10-31 19:54:23 0 [Note] Server socket created on IP: '0.0.0.0'.
2024-10-31 19:54:23 0 [Note] Server socket created on IP: '::'.
2024-10-31 19:54:23 0 [Note] mariadbd: Event Scheduler: Loaded 0 events
2024-10-31 19:54:23 3 [Note] Reading Master_info: 'master-mariadb@002doperator.info'  Relay_info:'relay-log-mariadb@002doperator.info'
2024-10-31 19:54:24 3 [Note] Initialized Master_info from 'master-mariadb@002doperator.info'
2024-10-31 19:54:24 4 [Note] Master 'mariadb-operator': Slave I/O thread: Start asynchronous replication to master 'repl@testrail-db-0.testrail-db-internal:3306' in log 'testrail-db-bin.000002' at position 285
2024-10-31 19:54:24 5 [Note] Master 'mariadb-operator': Slave SQL thread initialized, starting replication in log 'testrail-db-bin.000002' at position 285, relay log './testrail-db-relay-bin-mariadb@002doperator.000001' position: 4; GTID position '0-10-98'
2024-10-31 19:54:24 3 [Note] Started replication for 'mariadb-operator'
2024-10-31 19:54:24 0 [Note] mariadbd: ready for connections.
Version: '11.4.3-MariaDB-ubu2404-log'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
2024-10-31 19:54:24 4 [Note] Master 'mariadb-operator': Slave I/O thread: connected to master 'repl@testrail-db-0.testrail-db-internal:3306',replication starts at GTID position '0-10-98'
2024-10-31 19:54:39 5 [Note] Master 'mariadb-operator': Slave SQL thread exiting, replication stopped in log 'testrail-db-bin.000002' at position 85359954; GTID position '0-10-130', master: testrail-db-0.testrail-db-internal:3306
2024-10-31 19:55:47 4 [Note] Master 'mariadb-operator': Slave I/O thread exiting, read up to log 'testrail-db-bin.000002', position 566254795; GTID position 0-10-439, master testrail-db-0.testrail-db-internal:3306
2024-10-31 19:55:47 25 [Note] Master connection name: 'mariadb-operator'  Master_info_file: 'master-mariadb@002doperator.info'  Relay_info_file: 'relay-log-mariadb@002doperator.info'
2024-10-31 19:55:47 25 [Note] 'CHANGE MASTER TO executed'. Previous state master_host='testrail-db-0.testrail-db-internal', master_port='3306', master_log_file='testrail-db-bin.000002', master_log_pos='566254795'. New state master_host='testrail-db-0.testrail-db-internal', master_port='3306', master_log_file='', master_log_pos='4'.
2024-10-31 19:55:47 25 [Note] Previous Using_Gtid=Current_Pos. New Using_Gtid=Current_Pos
2024-10-31 19:55:48 38 [Note] Master 'mariadb-operator': Slave I/O thread: Start semi-sync replication to master 'repl@testrail-db-0.testrail-db-internal:3306' in log '' at position 4
2024-10-31 19:55:48 39 [Note] Master 'mariadb-operator': Slave SQL thread initialized, starting replication in log 'FIRST' at position 4, relay log './testrail-db-relay-bin-mariadb@002doperator.000001' position: 4; GTID position '0-10-130'
2024-10-31 19:55:48 38 [Note] Master 'mariadb-operator': Slave I/O thread: connected to master 'repl@testrail-db-0.testrail-db-internal:3306',replication starts at GTID position '0-10-130'
2024-10-31 19:57:18 60 [ERROR] Invalid (old?) table or database name 'lost+found'
```

**Comment by grooverdan** [comment]
So for clarity "Broken replication" is "Seconds_Behind_Master: 724090"? Was "Exec_Master_Log_Pos" moving in the `show all replica status\G`? It it was moving, but not fast enough - raise [slave_parallel_threads](https://mariadb.com/kb/en/replication-and-binary-log-system-variables/#slave_parallel_threads)

**Comment by sle78** [comment]
Hi, thanks for your reply, no it was not moving, I have fixed it by upgrading the operator from 0.0.33 to 0.35.1 through all the intermediate versions. Luckily the upgrade went smoothly and error free. Once on 0.35.1, I deleted the replica and its PVC and the operator did recover the cluster on its own automagically.
